### PR TITLE
LibWeb: Port IDL implementations to new String

### PR DESCRIPTION
**Pull request by kennethmyhra**
This ports the following IDL implementations to new String: `WorkerLocation`, `WorkerGlobalScope` and `Worker`.

BindingsGenerator has the following amendments:
- Allow stringifier to return `WebIDL::ExceptionOr<T>`
- Pass optional new String default value as a StringView to `String::from_utf8()`
- Release the value of Optional<String> when settings members of optional dictionary arguments to constructors

### CODE REVIEW

File: Meta/Lagom/Tools/CodeGenerators/LibWeb/BindingsGenerator/IDLGenerators.cpp [hunk 1]
```diff
@@ -729,9 +729,16 @@ static void generate_to_cpp(SourceGenerator& generator, ParameterType& parameter
                 }
 
                 generate_to_cpp(dictionary_generator, member, member_property_value_name, "", member_value_name, interface, member.extended_attributes.contains("LegacyNullToEmptyString"), !member.required, member.default_value);
-                dictionary_generator.append(R"~~~(
+                if (optional && interface.extended_attributes.contains("UseNewAKString")) {
```
Comment (AtkinsSJ) [inline Meta/Lagom/Tools/CodeGenerators/LibWeb/BindingsGenerator/IDLGenerators.cpp:732]: Commit message typo: "relase_value()"

File: Userland/Libraries/LibWeb/HTML/Worker.h [hunk 2]
```diff
@@ -26,18 +26,18 @@
 namespace Web::HTML {
 
 struct WorkerOptions {
-    DeprecatedString type { "classic" };
-    DeprecatedString credentials { "same-origin" };
-    DeprecatedString name { "" };
+    String type { String::from_utf8("classic"sv).release_value_but_fixme_should_propagate_errors() };
+    String credentials { String::from_utf8("same-origin"sv).release_value_but_fixme_should_propagate_errors() };
+    String name { String::from_utf8_short_string(""sv) };
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibWeb/HTML/Worker.h:31]: Any reason we can't just
```suggestion
    String name {};
```
? I think it's equivalent.

**Review by AtkinsSJ — CHANGES_REQUESTED** [review]
Very cool!

File: Userland/Libraries/LibWeb/HTML/Worker.h (hunk 2 above)
Comment (kennethmyhra) [reply Userland/Libraries/LibWeb/HTML/Worker.h]: None whatsoever, fixed:)

**Pull request merged**